File: Client/src/types/question.ts
import type { QuestionLevel } from '@/constants/questionLevels';

/** Lightweight question for lists (no answer body). */
export interface QuestionListItem {
    id: number | string;
    title: string;
    titleAr?: string | null;
    categoryId: number;
    subCategoryId: number;
    level?: QuestionLevel | null;
    isActive: boolean;
    createdAt: string;
    categoryName?: string;
    subCategoryName?: string;
}

/** id is number from API; route params may be string. */
export interface Question {
    id: number | string;
    /** Question title (English). */
    title: string;
    /** Question title (Arabic). Optional. */
    titleAr?: string | null;
    /** Answer in English. May contain Markdown. */
    answer: string;
    /** Answer in Arabic. Optional. May contain Markdown. */
    answerAr?: string | null;
    categoryId: number;
    subCategoryId: number;
    /** Seniority / difficulty level. */
    level?: QuestionLevel | null;
    isActive: boolean;
    createdAt: string;
    categoryName?: string;
    subCategoryName?: string;
}

export interface CreateQuestionDto {
    title: string;
    titleAr?: string | null;
    answer?: string;
    answerAr?: string | null;
    categoryId: number;
    subCategoryId: number;
    level: QuestionLevel;
    isActive?: boolean;
}

export interface UpdateQuestionDto {
    title: string;
    titleAr?: string | null;
    answer?: string;
    answerAr?: string | null;
    categoryId: number;
    subCategoryId: number;
    level: QuestionLevel;
    isActive: boolean;
}

File: Domain/Entities/Question.cs
namespace Domain.Entities;

public class Question
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string? TitleAr { get; set; }
    public string Answer { get; set; } = string.Empty;
    public string? AnswerAr { get; set; }
    public int CategoryId { get; set; }
    public int SubCategoryId { get; set; }
    public QuestionLevel? Level { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public Category Category { get; set; } = null!;
    public SubCategory SubCategory { get; set; } = null!;
}


public enum QuestionLevel
{
    Fresh = 0,
    Junior = 1,
    MidLevel = 2,
    Senior = 3
}
